lib/trie/database.go

```diff
@@ -145,24 +145,23 @@ func (t *Trie) loadProof(proofHashToNode map[string]Node, n Node) {
 
 // Load reconstructs the trie from the database from the given root hash.
 // It is used when restarting the node to load the current state trie.
-func (t *Trie) Load(db chaindb.Database, rootHash common.Hash) error {
-	if rootHash == EmptyHash {
+func (t *Trie) Load(db chaindb.Database, rootHashBytes []byte) error {
```

Comment:
How does this help?

I ran a tiny test

```go
func Test_ToBytes(t *testing.T) {
	h := Hash{1, 2, 3}
	t.Log(h)
	b := h.ToBytes()
	t.Log(b)
}
```

Printing

```
    /workspace/lib/common/hash_test.go:48: 0x0102030000000000000000000000000000000000000000000000000000000000
    /workspace/lib/common/hash_test.go:50: [1 2 3 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0]
```

So aren't these two the same?

Reply:
rootHash should always be hash I think (see above comment)?

Reply:
> How does this help?

It helped me to keep both the keys (when storing and loading) same. TBH, I did not consider that root node should always be hashed here. I think the right solution should have been something like https://github.com/ChainSafe/gossamer/pull/2359 instead of changing this arg. I am going to change this back to common.Hash.

Reply:
addressed